raft/raft.go

```diff
@@ -766,6 +766,29 @@ func (r *raft) becomeLeader() {
 	r.logger.Infof("%x became leader at term %d", r.id, r.Term)
 }
 
+func (r *raft) hup(t CampaignType) {
+	if r.state == StateLeader {
+		r.logger.Debugf("%x ignoring MsgHup because already leader", r.id)
+		return
+	}
+
+	if !r.promotable() {
+		r.logger.Warningf("%x is unpromotable and can not campaign; ignoring MsgHup", r.id)
+		return
+	}
+	ents, err := r.raftLog.slice(r.raftLog.applied+1, r.raftLog.committed+1, noLimit)
+	if err != nil {
+		r.logger.Panicf("unexpected error getting unapplied entries (%v)", err)
```

Comment:
Is there need `return err` ?

I can't pass `unit tests` with error
```
raft2020/07/23 19:40:17 unexpected error getting unapplied entries (requested index is unavailable due to compaction)
--- FAIL: TestLeaderTransferAfterSnapshot (0.00s)
panic: unexpected error getting unapplied entries (requested index is unavailable due to compaction) [recovered]
	panic: unexpected error getting unapplied entries (requested index is unavailable due to compaction)

goroutine 213 [running]:
testing.tRunner.func1.1(0xba71e0, 0xc000055bc0)
	/usr/local/go1.14.4/go/src/testing/testing.go:940 +0x421
testing.tRunner.func1(0xc000477680)
	/usr/local/go1.14.4/go/src/testing/testing.go:943 +0x600
panic(0xba71e0, 0xc000055bc0)
	/usr/local/go1.14.4/go/src/runtime/panic.go:975 +0x3e3
```

Reply:
@BusyJay Can you review this comment , It block my PR test ....

Reply:
This is an error that needs to be fixed by the removed commit. I will send it as a separate PR.

Reply:
PR is sent as #12163.